examples/step-7/step-7.cc

```diff
@@ -893,7 +893,10 @@ namespace Step7
     // from the L infinity errors on each cell with a call to
     // VectorTools::compute_global_error.
     const QTrapez<1>     q_trapez;
-    const QIterated<dim> q_iterated(q_trapez, 5);
+    const QIterated<dim> q_iterated(
+      q_trapez,
+      fe->degree * 2 +
+        1); // Not sure here if that is important, or if I can leave it at 5?
```

Comment:
This is probably a good choice you have here. `degree+3` would probably also have been ok. If you adjust it, please also adjust the comment right above with the expression you choose.

Reply:
When comparing the output, I get different convergence rates, even though that should not happen. For the original file, I get

```
Solving with Q1 elements, adaptive refinement
=============================================

Cycle 0:
   Number of active cells:       64
   Number of degrees of freedom: 81
Cycle 1:
   Number of active cells:       124
   Number of degrees of freedom: 157
Cycle 2:
   Number of active cells:       280
   Number of degrees of freedom: 341
Cycle 3:
   Number of active cells:       577
   Number of degrees of freedom: 690
Cycle 4:
   Number of active cells:       1099
   Number of degrees of freedom: 1264
Cycle 5:
   Number of active cells:       2191
   Number of degrees of freedom: 2452
Cycle 6:
   Number of active cells:       4165
   Number of degrees of freedom: 4510
Cycle 7:
   Number of active cells:       7915
   Number of degrees of freedom: 8440
Cycle 8:
   Number of active cells:       15196
   Number of degrees of freedom: 15912

cycle cells dofs     L2        H1      Linfty   
    0    64    81 1.576e-01 1.418e+00 2.707e-01 
    1   124   157 4.285e-02 1.285e+00 1.469e-01 
    2   280   341 1.593e-02 7.909e-01 8.034e-02 
    3   577   690 9.359e-03 5.096e-01 2.784e-02 
    4  1099  1264 2.865e-03 3.038e-01 9.822e-03 
    5  2191  2452 1.480e-03 2.106e-01 5.679e-03 
    6  4165  4510 6.907e-04 1.462e-01 2.338e-03 
    7  7915  8440 4.743e-04 1.055e-01 1.442e-03 
    8 15196 15912 1.920e-04 7.468e-02 7.259e-04 

Solving with Q1 elements, global refinement
===========================================

Cycle 0:
   Number of active cells:       64
   Number of degrees of freedom: 81
Cycle 1:
   Number of active cells:       256
   Number of degrees of freedom: 289
Cycle 2:
   Number of active cells:       1024
   Number of degrees of freedom: 1089
Cycle 3:
   Number of active cells:       4096
   Number of degrees of freedom: 4225
Cycle 4:
   Number of active cells:       16384
   Number of degrees of freedom: 16641

cycle cells dofs     L2        H1      Linfty   
    0    64    81 1.576e-01 1.418e+00 2.707e-01 
    1   256   289 4.280e-02 1.285e+00 1.444e-01 
    2  1024  1089 1.352e-02 7.556e-01 7.772e-02 
    3  4096  4225 3.423e-03 3.822e-01 2.332e-02 
    4 16384 16641 8.586e-04 1.917e-01 6.097e-03 

n cells         H1                  L2          
0    64 1.418e+00    -    - 1.576e-01    -    - 
1   256 1.285e+00 1.10 0.14 4.280e-02 3.68 1.88 
2  1024 7.556e-01 1.70 0.77 1.352e-02 3.17 1.66 
3  4096 3.822e-01 1.98 0.98 3.423e-03 3.95 1.98 
4 16384 1.917e-01 1.99 1.00 8.586e-04 3.99 2.00 

Solving with Q2 elements, global refinement
===========================================

Cycle 0:
   Number of active cells:       64
   Number of degrees of freedom: 289
Cycle 1:
   Number of active cells:       256
   Number of degrees of freedom: 1089
Cycle 2:
   Number of active cells:       1024
   Number of degrees of freedom: 4225
Cycle 3:
   Number of active cells:       4096
   Number of degrees of freedom: 16641
Cycle 4:
   Number of active cells:       16384
   Number of degrees of freedom: 66049

cycle cells dofs     L2        H1      Linfty   
    0    64   289 1.606e-01 1.278e+00 3.029e-01 
    1   256  1089 7.638e-03 5.248e-01 4.816e-02 
    2  1024  4225 8.601e-04 1.086e-01 4.827e-03 
    3  4096 16641 1.107e-04 2.756e-02 7.802e-04 
    4 16384 66049 1.393e-05 6.915e-03 9.971e-05 

n cells         H1                   L2          
0    64 1.278e+00    -    - 1.606e-01     -    - 
1   256 5.248e-01 2.43 1.28 7.638e-03 21.03 4.39 
2  1024 1.086e-01 4.83 2.27 8.601e-04  8.88 3.15 
3  4096 2.756e-02 3.94 1.98 1.107e-04  7.77 2.96 
4 16384 6.915e-03 3.99 1.99 1.393e-05  7.94 2.99 

Solving with Q2 elements, adaptive refinement
===========================================

Cycle 0:
   Number of active cells:       64
   Number of degrees of freedom: 289
Cycle 1:
   Number of active cells:       124
   Number of degrees of freedom: 577
Cycle 2:
   Number of active cells:       289
   Number of degrees of freedom: 1353
Cycle 3:
   Number of active cells:       547
   Number of degrees of freedom: 2531
Cycle 4:
   Number of active cells:       1057
   Number of degrees of freedom: 4919
Cycle 5:
   Number of active cells:       2059
   Number of degrees of freedom: 9223
Cycle 6:
   Number of active cells:       3913
   Number of degrees of freedom: 17887
Cycle 7:
   Number of active cells:       7441
   Number of degrees of freedom: 33807
Cycle 8:
   Number of active cells:       14212
   Number of degrees of freedom: 64731

cycle cells dofs     L2        H1      Linfty   
    0    64   289 1.606e-01 1.278e+00 3.029e-01 
    1   124   577 7.891e-03 5.256e-01 4.852e-02 
    2   289  1353 1.070e-03 1.155e-01 4.868e-03 
    3   547  2531 5.962e-04 5.101e-02 1.876e-03 
    4  1057  4919 1.977e-04 3.094e-02 7.923e-04 
    5  2059  9223 7.738e-05 1.974e-02 7.270e-04 
    6  3913 17887 2.925e-05 8.772e-03 1.463e-04 
    7  7441 33807 1.024e-05 4.121e-03 8.567e-05 
    8 14212 64731 3.761e-06 2.108e-03 2.167e-05 
```

and for the modified file I get

```
Solving with Q1 elements, adaptive refinement
=============================================

Cycle 0:
   Number of active cells:       64
   Number of degrees of freedom: 81
Cycle 1:
   Number of active cells:       124
   Number of degrees of freedom: 157
Cycle 2:
   Number of active cells:       280
   Number of degrees of freedom: 341
Cycle 3:
   Number of active cells:       577
   Number of degrees of freedom: 690
Cycle 4:
   Number of active cells:       1099
   Number of degrees of freedom: 1264
Cycle 5:
   Number of active cells:       2191
   Number of degrees of freedom: 2452
Cycle 6:
   Number of active cells:       4165
   Number of degrees of freedom: 4510
Cycle 7:
   Number of active cells:       7915
   Number of degrees of freedom: 8440
Cycle 8:
   Number of active cells:       15196
   Number of degrees of freedom: 15912

cycle cells dofs     L2        H1      Linfty   
    0    64    81 1.840e+00 2.858e+00 1.835e+00 
    1   124   157 5.190e-02 1.200e+00 1.344e-01 
    2   280   341 1.439e-02 7.892e-01 7.554e-02 
    3   577   690 8.627e-03 5.061e-01 2.805e-02 
    4  1099  1264 3.217e-03 3.030e-01 1.073e-02 
    5  2191  2452 1.445e-03 2.097e-01 5.073e-03 
    6  4165  4510 8.387e-04 1.460e-01 2.013e-03 
    7  7915  8440 7.051e-04 1.053e-01 1.804e-03 
    8 15196 15912 2.774e-04 7.463e-02 6.911e-04 

Solving with Q1 elements, global refinement
===========================================

Cycle 0:
   Number of active cells:       64
   Number of degrees of freedom: 81
Cycle 1:
   Number of active cells:       256
   Number of degrees of freedom: 289
Cycle 2:
   Number of active cells:       1024
   Number of degrees of freedom: 1089
Cycle 3:
   Number of active cells:       4096
   Number of degrees of freedom: 4225
Cycle 4:
   Number of active cells:       16384
   Number of degrees of freedom: 16641

cycle cells dofs     L2        H1      Linfty   
    0    64    81 1.840e+00 2.858e+00 1.835e+00 
    1   256   289 3.570e-02 1.199e+00 1.307e-01 
    2  1024  1089 1.192e-02 7.565e-01 7.168e-02 
    3  4096  4225 3.047e-03 3.823e-01 2.128e-02 
    4 16384 16641 7.660e-04 1.917e-01 5.554e-03 

n cells         H1                   L2          
0    64 2.858e+00    -    - 1.840e+00     -    - 
1   256 1.199e+00 2.38 1.25 3.570e-02 51.54 5.69 
2  1024 7.565e-01 1.58 0.66 1.192e-02  2.99 1.58 
3  4096 3.823e-01 1.98 0.98 3.047e-03  3.91 1.97 
4 16384 1.917e-01 1.99 1.00 7.660e-04  3.98 1.99 

Solving with Q2 elements, global refinement
===========================================

Cycle 0:
   Number of active cells:       64
   Number of degrees of freedom: 289
Cycle 1:
   Number of active cells:       256
   Number of degrees of freedom: 1089
Cycle 2:
   Number of active cells:       1024
   Number of degrees of freedom: 4225
Cycle 3:
   Number of active cells:       4096
   Number of degrees of freedom: 16641
Cycle 4:
   Number of active cells:       16384
   Number of degrees of freedom: 66049

cycle cells dofs     L2        H1      Linfty   
    0    64   289 1.606e-01 1.278e+00 3.029e-01 
    1   256  1089 7.638e-03 5.248e-01 4.816e-02 
    2  1024  4225 8.601e-04 1.086e-01 4.827e-03 
    3  4096 16641 1.107e-04 2.756e-02 7.802e-04 
    4 16384 66049 1.393e-05 6.915e-03 9.971e-05 

n cells         H1                   L2          
0    64 1.278e+00    -    - 1.606e-01     -    - 
1   256 5.248e-01 2.43 1.28 7.638e-03 21.03 4.39 
2  1024 1.086e-01 4.83 2.27 8.601e-04  8.88 3.15 
3  4096 2.756e-02 3.94 1.98 1.107e-04  7.77 2.96 
4 16384 6.915e-03 3.99 1.99 1.393e-05  7.94 2.99 

Solving with Q2 elements, adaptive refinement
===========================================

Cycle 0:
   Number of active cells:       64
   Number of degrees of freedom: 289
Cycle 1:
   Number of active cells:       124
   Number of degrees of freedom: 577
Cycle 2:
   Number of active cells:       289
   Number of degrees of freedom: 1353
Cycle 3:
   Number of active cells:       547
   Number of degrees of freedom: 2531
Cycle 4:
   Number of active cells:       1057
   Number of degrees of freedom: 4919
Cycle 5:
   Number of active cells:       2059
   Number of degrees of freedom: 9223
Cycle 6:
   Number of active cells:       3913
   Number of degrees of freedom: 17887
Cycle 7:
   Number of active cells:       7441
   Number of degrees of freedom: 33807
Cycle 8:
   Number of active cells:       14212
   Number of degrees of freedom: 64731

cycle cells dofs     L2        H1      Linfty   
    0    64   289 1.606e-01 1.278e+00 3.029e-01 
    1   124   577 7.891e-03 5.256e-01 4.852e-02 
    2   289  1353 1.070e-03 1.155e-01 4.868e-03 
    3   547  2531 5.962e-04 5.101e-02 1.876e-03 
    4  1057  4919 1.977e-04 3.094e-02 7.923e-04 
    5  2059  9223 7.738e-05 1.974e-02 7.270e-04 
    6  3913 17887 2.925e-05 8.772e-03 1.463e-04 
    7  7441 33807 1.024e-05 4.121e-03 8.567e-05 
    8 14212 64731 3.761e-06 2.108e-03 2.167e-05 
```

Reply:
Comparing the data, the only thing that looks like it changes is really the Linfty norm, right? For that one I'm not at all surprised -- you're trying to find that one point where the error is largest, and that will necessarily depend on where exactly your quadrature points are.

Or am I missing anything?

Reply:
The L2-norm for the Q1-elements also changes, due to the change from `QGauss<dim>(3)` to effectively `QGauss<dim>(2)`.

Reply:
Ah right. But it seems to be a constant factor, the convergence rates are still the same. So I would still say to just go with the "corrected" version and just update the output.